ddtrace/contrib/botocore/patch.py

```diff
@@ -61,28 +61,41 @@ class TraceInjectionDecodingError(Exception):
     pass
 
 
-def inject_trace_data_to_message_attributes(trace_data, entry):
-    # type: (Dict[str, str], Dict[str, Any]) -> None
+def inject_trace_data_to_message_attributes(trace_data, entry, endpoint):
```

Comment:
Adding a parameter to a public function is a breaking change and would require a major version bump (ie releasing this change in v1.0 or v2.0). We could add a default argument here instead. 

```suggestion
def inject_trace_data_to_message_attributes(trace_data, entry, endpoint = ""):
```

@brettlangdon thoughts?

Reply:
🤔 good question. technically, yes this is a public function.

In that case, `endpoint=None` with typing of `Optional[Union[Literal["sqs"], Literal["sns"]]]` would be best ?

Reply:
updated in [77afe63](https://github.com/DataDog/dd-trace-py/pull/3404/commits/77afe63a59d892cd7751f183f242491e21207ced)

Are all the other functions that use endpoint public as well? I.e. should I do this with the other functions, `inject_trace_to_sqs_or_sns_batch_message` and `inject_trace_to_sqs_or_sns_message`. Also what about`patched_api_call`?

Reply:
Anything that isn't prefixed with a `_` technically is considered part of our public API that customers could be importing and using.